### PR TITLE
Add AnimalCommandTestUtil and NameContainsKeywordsPredicate

### PR DESCRIPTION
**Pull request by junnengsoo**
AnimalCommandTestUtil has attributes that are still unmodified from Person model and NameContainsKeywordsPredicate was added to the animal model directory with no test coverage yet.

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/AnimalCommandTestUtil.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+    static {
+        DESC_AMY = new EditPersonDescriptorBuilder().withName(VALID_NAME_AMY)
+                .withPhone(VALID_PHONE_AMY).withEmail(VALID_EMAIL_AMY).withAddress(VALID_ADDRESS_AMY)
+                .withTags(VALID_TAG_FRIEND).build();
+        DESC_BOB = new EditPersonDescriptorBuilder().withName(VALID_NAME_BOB)
+                .withPhone(VALID_PHONE_BOB).withEmail(VALID_EMAIL_BOB).withAddress(VALID_ADDRESS_BOB)
+                .withTags(VALID_TAG_HUSBAND, VALID_TAG_FRIEND).build();
+    }
+
```
Comment (wjayee) [inline src/test/java/seedu/address/logic/commands/AnimalCommandTestUtil.java:71]: Should this refer to animal and its attributes instead of person?

**Review by wjayee — CHANGES_REQUESTED** [review]
LGTM! Just a question about the AnimalCommandTestUtil file

File: src/main/java/seedu/address/model/animal/NameContainsKeywordsPredicate.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    private final List<String> keywords;
+
+    public NameContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Animal animal) {
+        return keywords.stream()
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(animal.getName().fullName, keyword));
```
Comment (euchangxian) [inline src/main/java/seedu/address/model/animal/NameContainsKeywordsPredicate.java:22]: Do add JavaDocs describing the behaviour of this method, specifically when using `Stream::anyMatch`. If I recall correctly, it returns any words that contain a subset of the specified string.

File: src/main/java/seedu/address/model/animal/NameContainsKeywordsPredicate.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof NameContainsKeywordsPredicate)) {
+            return false;
+        }
+
+        NameContainsKeywordsPredicate otherNameContainsKeywordsPredicate = (NameContainsKeywordsPredicate) other;
+        return keywords.equals(otherNameContainsKeywordsPredicate.keywords);
```
Comment (euchangxian) [inline src/main/java/seedu/address/model/animal/NameContainsKeywordsPredicate.java:37]: Do add a comment explaining why this is a safe-cast.

**Review by euchangxian — APPROVED** [review]
LGTM with a few nits. Do look at the comments and address them.

**Review by wjayee — APPROVED** [review]

**Pull request merged**